sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/SqlTransform.java

```diff
@@ -103,6 +105,8 @@
 
   abstract @Nullable String defaultTableProvider();
 
+  abstract @Nullable String queryPlannerClassName();
```

Comment:
Can you add some comments here or in class comment so clarify the ordering of getting planner names? E.g. get the class name set in SqlTransform first and the try pipeline options.